src/tutorial.rs

```diff
@@ -0,0 +1,143 @@
+use std::sync::Mutex;
+
+pub struct Tutorial {
+    current_step: Mutex<i8>,
+    num_steps: i8,
+}
+
+impl Tutorial {
+    pub fn new() -> Self {
+        let tutorial = Self {
+            current_step: Mutex::new(0),
+            num_steps: (STEPS.len() as i8) - 1,
+        };
+
+        tutorial
+    }
+
+    pub fn current_step(&self) -> &str {
+        let step = *self.current_step.lock().unwrap();
+        STEPS[step as usize].clone()
+    }
+
+    pub fn next_tutorial_step(&self) -> &str {
+        let mut current_step = self.current_step.lock().unwrap();
+        *current_step += 1;
+        if *current_step > self.num_steps {
+            *current_step = 0;
+        }
+        STEPS[*current_step as usize].clone()
+    }
+
+    pub fn prev_tutorial_step(&self) -> &str {
+        let mut current_step = self.current_step.lock().unwrap();
+        *current_step -= 1;
+        if *current_step < 0 {
+            *current_step = self.num_steps - 1;
+        }
+        STEPS[*current_step as usize].clone()
+    }
+}
+
+static STEPS: &'static [&'static str] = &[
```

Comment:
what if we make this a Map of str str ? The name could then be used to list all the steps in a table when doing something like "tutorial steps" ? .. I could imagine someone want to jump to a specific tutorial step like i.e. query or something